app/services/evaluation_status_service.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+# This service handles calculating the overall evaluation status of a submission
+class EvaluationStatusService
+  def self.calculate_evaluation_status(submission)
+    new(submission).calculate_evaluation_status
+  end
```

Comment:
Service updates the model directly
```suggestion
  def self.update_evaluation_status(submission)
    evaluation_status = new(submission).calculate_evaluation_status
    submission.update(evaluation_status:)
  end
```